File: check_images.js
const fs = require('fs');
const https = require('https');

const urls = fs.readFileSync('constants.ts', 'utf8')
  .match(/https:\/\/images\.unsplash\.com\/[a-zA-Z0-9-]+\?auto=format&fit=crop&q=80&w=1200/g)
  || [];

const uniqueUrls = [...new Set(urls)];
console.log(`Checking ${uniqueUrls.length} images...`);

uniqueUrls.forEach(url => {
  https.request(url, { method: 'HEAD' }, res => {
    if (res.statusCode >= 400) {
      console.log(`BROKEN: ${url} (Status: ${res.statusCode})`);
    }
  }).on('error', err => {
    console.log(`ERROR: ${url} (${err.message})`);
  }).end();
});

File: check_images.py
import urllib.request

with open('constants.ts', 'r') as f:
    content = f.read()

import re
urls = re.findall(r"'(https://images.unsplash.com[^']+)'", content)
urls = list(set(urls))
for url in urls:
    try:
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        res = urllib.request.urlopen(req)
        if res.status != 200:
            print(f"Error {res.status}: {url}")
    except Exception as e:
        print(f"Failed {url}: {e}")
